tools/etw-provider-gen/main.go

```diff
@@ -0,0 +1,36 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"os"
+	"syscall"
+	"unsafe"
+
+	"github.com/Microsoft/go-winio/internal/etw"
+)
+
+func main() {
+	var pn = flag.String("provider-name", "", "The human readable ETW provider name to be converted into GUID format")
+	flag.Parse()
+	if pn == nil || *pn == "" {
+		fmt.Fprint(os.Stderr, "--provider-name is required")
+		os.Exit(1)
+	}
+	p, err := etw.NewProvider(*pn, nil)
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "failed to convert provider-name: '%s' with err: '%s", *pn, err)
+		os.Exit(1)
+	}
+	defer p.Close()
+
+	var resultp *uint16
+	err = stringFromCLSID(p.ID, &resultp)

```

Comment:
You could also use the guidToString function I wrote here: https://github.com/Microsoft/go-winio/blob/master/internal/etw/sample/sample.go#L21

Reply:
Deal